File: 2491-divide-players-into-teams-of-equal-skill/2491-divide-players-into-teams-of-equal-skill.py
class Solution:
    def dividePlayers(self, skill: List[int]) -> int:
        n, sumt = len(skill)//2, sum(skill)
        if sumt%n!=0: return -1
        
        skill.sort()
        team, ans = sumt//n, 0
        for i in range(n):
            if skill[i]+skill[2*n-i-1] != team: return -1
            ans += skill[i]*skill[2*n-i-1]
        return ans
        

File: 2491-divide-players-into-teams-of-equal-skill/2491-divide-players-into-teams-of-equal-skill.java
class Solution {
    public long dividePlayers(int[] skill) {
        Arrays.sort(skill);
        int n = skill.length;
        int sumt = skill[0]+skill[n-1];
        long ans = 0;
        for(int i=0; i<n/2; ++i){
            if(skill[i]+skill[n-i-1] != sumt) return -1;
            else ans += (long)skill[i] * (long)skill[n-i-1];
        }
        return ans;
    }
}